(x3270): packaged x3270 IBM terminal emulator

musteresel opened the pull request:
I added a nix expression to build the x3270 IBM 3270 terminal emulator, to be found there:

http://x3270.bgp.nu/index.html

I packaged the current stable version (3.4ga9 stable (23. October 2015), according to their downloads page). Though I pull the source code from the git repository instead of the provided tar balls. A small patch to fix the interpreter of an utility bash script was needed.

Actually this package now provides multiple binaries. One could add support to only build a subset of these binaries (and thus removing tcl, x dependencies). Also there are no man pages, because the respective install target fails and I don't have the time to investigate further.

As for testing I'm successfully running the applications on my laptop (NixOS on x86_64), using them to participate in the IBM Master the Mainframe contest. No further testing than this was done, though.

Please let me know if there's something that needs to be changed (this is my first to be contributed package).

rickynils commented:
@musteresel Could you squash your commits into a single one, and set the commit message according to the style defined in http://nixos.org/nixpkgs/manual/#chap-submitting-changes?

dezgeg commented:
Also this seems to be targeting the release branch, this should go to master first and only then cherry-picked to `release-15.09`.

musteresel committed:
x3270: init at 3.4ga9

x3270 is an IBM 3270 terminal emulator for the X Window System. It comes
with some other - related - tools, such as a ncurses version or a
version intended for use with scripts. These tools are currently
unconditionally built, this is a possible point of improvement. Also the
man pages unfortunately don't install with this nix expression.

Version 3.4ga9 is the current (23. Oct 2015) stable version of this
software.

musteresel commented:
Squashing into a single commit and formatting that according to the rules worked as expected. Rebase to the master branch worked, too, but github didn't get it, so I create a new pull request #11080
